htdocs/core/lib/files.lib.php

```diff
@@ -2992,7 +2992,7 @@ function dol_check_secure_access_document($modulepart, $original_file, $entity,
 			if ($partofdirinoriginalfile && ($fuser->rights->$modulepart->$partofdirinoriginalfile->{$lire} || $fuser->rights->$modulepart->$partofdirinoriginalfile->{$read})) $accessallowed = 1;
 			if ($fuser->rights->$modulepart->{$lire} || $fuser->rights->$modulepart->{$read}) $accessallowed = 1;
 			//}
-			$original_file = $conf->$modulepart->dir_output.'/'.$original_file;
+			$original_file = $conf->$modulepart->multidir_output[$entity].'/'.$original_file;
```

Comment:
Are you sure $conf->$modulepart->multidir_output[$entity] is always defined ?
May be a fallback on old path $conf->$modulepart->dir_output is more secured ?

Reply:
Normally always set with multicompnay, but your right as multicompany is not a internal module better to be secure and deal with allbask.